packages/nodes-from-anchor/src/v01/PdaNode.ts

```diff
@@ -1,39 +1,102 @@
-import { KINOBI_ERROR__ANCHOR__UNRECOGNIZED_IDL_TYPE, KinobiError } from '@kinobi-so/errors';
+import {
+    KINOBI_ERROR__ANCHOR__ACCOUNT_TYPE_MISSING,
+    KINOBI_ERROR__ANCHOR__ARGUMENT_TYPE_MISSING,
+    KINOBI_ERROR__ANCHOR__FIELD_TYPE_MISSING,
+    KinobiError,
+} from '@kinobi-so/errors';
 import {
     bytesTypeNode,
     camelCase,
     constantPdaSeedNode,
     fixedSizeTypeNode,
     PdaNode,
     pdaNode,
-    PdaSeedNode,
     publicKeyTypeNode,
     variablePdaSeedNode,
 } from '@kinobi-so/nodes';
 
 import { getAnchorDiscriminatorV01 } from './../discriminators';
-import { IdlV01InstructionAccount } from './idl';
-
-export function pdaNodeFromAnchorV01(idl: IdlV01InstructionAccount): PdaNode {
-    const seeds = idl.pda?.seeds.map((seed): PdaSeedNode => {
-        switch (seed.kind) {
-            case 'const':
-                return constantPdaSeedNode(
-                    fixedSizeTypeNode(bytesTypeNode(), seed.value.length),
-                    getAnchorDiscriminatorV01(seed.value),
-                );
-            case 'account':
-                return variablePdaSeedNode(seed.path, publicKeyTypeNode());
-            case 'arg':
-                throw new KinobiError(KINOBI_ERROR__ANCHOR__UNRECOGNIZED_IDL_TYPE, { idlType: seed });
-        }
-    });
+import {
+    IdlV01Field,
+    IdlV01Instruction,
+    IdlV01InstructionAccount,
+    IdlV01Seed,
+    IdlV01TypeDef,
+    IdlV01TypeDefTyStruct,
+} from './idl';
+import { typeNodeFromAnchorV01 } from './typeNodes';
+
+export function pdaNodesFromAnchorInstructionV01(instruction: IdlV01Instruction, types: IdlV01TypeDef[]): PdaNode[] {
+    return instruction.accounts.reduce((memo: PdaNode[], account: IdlV01InstructionAccount) => {
+        const pda = account.pda;
+        if (pda) {
+            const name = camelCase(`${instruction.name}_${account.name}`);
```

Comment:
> Also note that we could update the PdaValueNode such that its pda attribute could either be a PdaLinkNode or a PdaNode directly. That way each instruction account would be responsible for defining its own PdaNode. Then, we can use the defaultVisitor to goes through each of these inlined PdaNode and transform them into PdaLinkNode when applicable.

I'm up to explore this approach. At least for how anchor defines pda they shouldn't be directly exposed to the end caller. The IDL allows for the seeds to be computed off of the context of the instruction. I'm actually thinking that they shouldn't be included in Kinobi program pda list as they are helps generally defined on account independently of the instruction.

Reply:
It makes sense, I don't think it would be that big of a change and it wouldn't even be a breaking change. I can have a look at it tomorrow by working on your branch. I'll also need to add an optional `programId` attribute to the `PdaNode` so it can derive from other programs too.

I still think it could be valuable to have a visitor that goes through the whole tree afterwards and try to extract some of these nested `PdaNodes` into top-level `PdaNodes` so the end-user has helpers to create these PDAs. We would just need to be extra-strict on the ones we extract. For instance, we would only extract the ones that are from this program and we would make sure we deduplicate any PDA that's used in multiple instructions. But the beauty of having that on its own visitor is people can opt-out of this whole logic by opting out of the default visitors.